redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,678 @@
+                messages::serviceTemporarilyUnavailable(asyncResp->res, "60");
+                return;
+            }
+        };
+
+        timeout->async_wait(timeoutHandler);
+
+        auto callback = [asyncResp, timeout,
+                         licenseString](sdbusplus::message::message& m) {
+            BMCWEB_LOG_DEBUG("Response Matched ");
+            boost::container::flat_map<std::string, std::variant<std::string>>
+                values;
+            std::string iface;
+            m.read(iface, values);
+            if (iface == "com.ibm.License.LicenseManager")
+            {
+                auto findStatus = values.find("LicenseActivationStatus");
+                if (findStatus != values.end())
+                {
+                    BMCWEB_LOG_INFO("Found Status property change");
+                    std::string* status =
+                        std::get_if<std::string>(&(findStatus->second));
+                    if (status == nullptr)
+                    {
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+                    // Ignore D-bus propertyChanged signal for
+                    // status value "Pending"
+                    if (*status == "com.ibm.License.LicenseManager."
+                                   "Status.Pending")
+                    {
+                        return;
+                    }
+                    getLicenseActivationAck(asyncResp, *status, licenseString);
+                    timeout->cancel();
+                }
+            }
+        };
+
+        getLicenseActivationStatusMatch() =
+            std::make_unique<sdbusplus::bus::match::match>(
+                *crow::connections::systemBus,
+                "interface='org.freedesktop.DBus.Properties',type='"
+                "signal',"
+                "member='PropertiesChanged',path='/com/ibm/license'",
+                callback);
+    },
+        "com.ibm.License.Manager", "/com/ibm/license",
+        "org.freedesktop.DBus.Properties", "Set",
+        "com.ibm.License.LicenseManager", "LicenseString",
+        std::variant<std::string>(licenseString));
+}
+inline void requestRoutesLicenseEntryCollection(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/LicenseService/Licenses")
+        .privileges({{"Login"}})
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp) {
+        asyncResp->res.jsonValue["@odata.type"] =
+            "#LicenseCollection.LicenseCollection";
+        asyncResp->res.jsonValue["@odata.id"] =
+            "/redfish/v1/LicenseService/Licenses";
+        asyncResp->res.jsonValue["Name"] = "License Collection";
+
+        getLicenseEntryCollection(asyncResp);
+    });
+
+    BMCWEB_ROUTE(app, "/redfish/v1/LicenseService/Licenses")
+        .privileges({{"ConfigureManager"}})
```

Comment:
PatchLicenseCollection

Reply:
what you mean by PatchLicenseCollection?

Reply:
https://github.com/ibm-openbmc/bmcweb/blob/1060/redfish-core/lib/ethernet.hpp#L2281

Reply:
But crap you need https://gerrit.openbmc.org/c/openbmc/bmcweb/+/70854

Reply:
If you want to rebase and use, I think it is better but I won't require it

Reply:
I will not do it as part of this PR. Will add separate line item for it